File: PlacementCellBackend/Services/Placements/Interfaces/IAlumniPlacementService.cs
using PlacementCellBackend.DTOs.AlumniJobOpenings;

namespace PlacementCellBackend.Services.Placements.Interfaces
{
    public interface IAlumniPlacementService
    {
        Task<IEnumerable<AlumniJobOpeningDto>> GetAllAlumniJobPostionAsync();
        Task<AlumniJobOpeningDto?> GetAlumniPlacementByIdAsync(int id);
        Task<bool> CreateAlumniPlacementAsync(AlumniJobOpeningCreateDto alumniPlacement);
        Task<bool> UpDateAlumniPlacementAsync(int id, AlumniJobOpeningCreateDto alumniPlacement);
        Task<bool> DeleteAlumniPlacementAsync(int id);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Placements/Interfaces/IAlumniPlacementService.java
package com.example.placementicsbackend.services.placements.interfaces;

import com.example.placementicsbackend.dtos.AlumniJobOpenings.AlumniJobOpeningCreateDto;
import com.example.placementicsbackend.dtos.AlumniJobOpenings.AlumniJobOpeningDto;
import java.util.List;

public interface IAlumniPlacementService {

    List<AlumniJobOpeningDto> getAllAlumniJobPositions();

    AlumniJobOpeningDto getAlumniPlacementById(int id);

    boolean createAlumniPlacement(AlumniJobOpeningCreateDto alumniPlacement);

    boolean updateAlumniPlacement(int id, AlumniJobOpeningCreateDto alumniPlacement);

    boolean deleteAlumniPlacement(int id);
}
